### PR TITLE
Registry Updates

### PR DESCRIPTION
**Pull request by jalextowle**
# Description

This PR updates the registry to support deterministic deployments using CREATE3.

### CODE REVIEW

File: test/units/registry/HyperdriveRegistry.t.sol [hunk 1]
```diff
@@ -32,14 +32,20 @@ contract HyperdriveRegistryTest is HyperdriveTest {
     string internal constant REGISTRY_NAME = "HyperdriveRegistry";
     uint256 internal constant FIXED_RATE = 0.05e18;
 
+    event Initialized(string indexed name, address indexed admin);
```
Comment (mcclurejt) [inline test/units/registry/HyperdriveRegistry.t.sol:35]: Why not use an events interfact similar to IHyperdriveEvents?

File: .env_template [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
 
-# private key used for running migration scripts
-PRIVATE_KEY=
+# The environment variables used in the deployment scripts.
+HYPERDRIVE_ETHEREUM_URL=
+DEPLOYER_PRIVATE_KEY=
+PAUSER_PRIVATE_KEY=
 
-# API key used to verify contracts
+# The API key used to verify contracts.
```
Comment (mcclurejt) [inline .env_template:15]:
```suggestion
# The API keys used to verify contracts.
```

File: .env_template [hunk 3]
```diff
@@ -1,17 +1,23 @@
-# ethereum rpc endpoints used in fork tests and migration scripts
+# The ethereum rpc endpoints used in fork tests and migration scripts.
```
Comment (mcclurejt) [inline .env_template:1]:
```suggestion
# The rpc endpoints used in fork tests and migration scripts.
```

**Review by mcclurejt — APPROVED** [review]
👍

File: contracts/src/registry/HyperdriveRegistry.sol [hunk 4]
```diff
@@ -68,11 +64,33 @@ contract HyperdriveRegistry is
...
 
     /// @inheritdoc IHyperdriveGovernedRegistry
-    function updateAdmin(address _admin) external override onlyAdmin {
+    function initialize(string calldata _name, address _admin) external {
+        // Ensure that the registry hasn't already been initialized.
+        if (isInitialized) {
+            revert IHyperdriveGovernedRegistry.RegistryAlreadyInitialized();
+        }
+
+        // Set the initialization flag and set the name and admin.
```
Comment (jrhea) [inline contracts/src/registry/HyperdriveRegistry.sol:73]:
```suggestion
        // Set the initialization flag, name and admin.
```

**Review by jrhea — APPROVED** [review]

**Pull request merged**